protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpSession.java

```diff
@@ -279,22 +277,12 @@ public void send(Packet packet) {
     }
 
     @Override
-    public void disconnect(String reason) {
-        this.disconnect(Component.text(reason));
-    }
-
-    @Override
-    public void disconnect(String reason, Throwable cause) {
-        this.disconnect(Component.text(reason), cause);
-    }
-
-    @Override
-    public void disconnect(Component reason) {
+    public void disconnect(@NonNull Component reason) {
         this.disconnect(reason, null);
     }
 
     @Override
-    public void disconnect(final Component reason, final Throwable cause) {
+    public void disconnect(@NonNull Component reason, @Nullable Throwable cause) {
```

Comment:
We use checkerframework's NonNull/Nullable annotations in other projects as a standard; let's also do so here?

Reply:
But packets have lomboks NonNull. But sure I'll do it.

Reply:
We'll probably want to change that as well in the future :p

Reply:
The nice thing about lomboks annotations is that they generate enforced null checks at compile time.